deps/v8/src/compiler/instruction.cc

```diff
@@ -683,43 +683,32 @@ FrameStateDescriptor::FrameStateDescriptor(
 }
 
 
-size_t FrameStateDescriptor::GetSize(OutputFrameStateCombine combine) const {
-  size_t size = 1 + parameters_count() + locals_count() + stack_count() +
-                (HasContext() ? 1 : 0);
-  switch (combine.kind()) {
-    case OutputFrameStateCombine::kPushOutput:
-      size += combine.GetPushCount();
-      break;
-    case OutputFrameStateCombine::kPokeAt:
-      break;
-  }
-  return size;
-}
-
-
-size_t FrameStateDescriptor::GetTotalSize() const {
+size_t FrameStateDescriptor::GetTotalSize(const FrameStateDescriptor* desc) {
+  if (desc == NULL) return 0;
```

Comment:
This nullptr check is not strictly necessary.  Same for the corresponding checks in the other methods.